Auth0/WebViewProvider.swift

```diff
@@ -0,0 +1,163 @@
+//
+//  WebViewProvider.swift
+//  Auth0
+//
+//  Created by Desu Sai Venkat on 18/09/24.
+//  Copyright © 2024 Auth0. All rights reserved.
+//
+
+#if os(iOS)
+
+@preconcurrency import WebKit
+
+
+extension WebAuthentication {
+    static func webViewProvider(redirectionURL: URL, style: UIModalPresentationStyle = .fullScreen) -> WebAuthProvider {
+        return { url, callback  in
+            WebViewUserAgent(authorizeURL: url, redirectURL: redirectionURL, modalPresentationStyle: style, callback: callback)
+        }
+    }
+}
+
+extension WebViewUserAgent {
+    var topViewController: UIViewController? {
```

Comment:
We already have this logic here: https://github.com/auth0/Auth0.swift/blob/master/Auth0/SafariProvider.swift#L50-L77
Maybe we can put it in some kind of shared helper (e.g. an internal static var added to `UIWindow` –analogous to `rootViewController`)?

Reply:
agreed, moved this to a shared helper